### PR TITLE
Add support for teams, and add tests

### PR DESCRIPTION
**Pull request by markelliot**
## Before this PR
No support for teams, no tests.

## After this PR
Add support for teams and add missing tests.

### CODE REVIEW

**Comment by markelliot** [comment]
couple of notes: 
- we're using a vendored yarn cache (part of yarn v2)
- because this is a GitHub action we're currently updating dist/ with the build result to make the action releasable

I think in the future we can kill the second thing by pushing dist/ + action.yml to a separate branch and tagging that branch rather than develop (and we can do that in an automated way), but will come back to do that on a separate pass.

I'll also wipe out the lib/ directory with a .gitignore on another pass.

File: dist/index.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ var __importStar = (this && this.__importStar) || function (mod) {
     return result;
 };
 Object.defineProperty(exports, "__esModule", ({ value: true }));
+exports.check = void 0;
```
Comment (ferozco) [inline dist/index.js:33]: should this be checked in? seems like a duplication of `main/index.js`

File: lib/main.js [hunk 2]
```diff
@@ -23,6 +23,7 @@ var __importStar = (this && this.__importStar) || function (mod) {
     return result;
 };
 Object.defineProperty(exports, "__esModule", { value: true });
+exports.check = void 0;
```
Comment (ferozco) [inline lib/main.js:26]: same here seems weird to have these checked in

File: src/main.ts [hunk 3]
```diff
@@ -26,6 +35,69 @@ function getPrNumber(context: Context): number | undefined {
   return undefined;
 }
 
+function getPossibleApprovers(
+  conf: ReviewerConfiguration,
+  teams: { [key: string]: TeamConfiguration }
+): Set<string> {
+  const namedUsers = conf.users || [];
+  const usersFromAllNamedTeams = (conf.teams || [])
+    .map((team) => teams[team].users)
```
Comment (ferozco) [inline src/main.ts:44]: you ca just use `.flatMap` instead

**Review by ferozco — APPROVED** [review]

**Review by ferozco — APPROVED** [review]

File: dist/index.js (hunk 1 above)
Comment (markelliot) [reply dist/index.js:33]: see my top-level comment

File: src/main.ts (hunk 3 above)
Comment (markelliot) [reply src/main.ts:44]: I was not able to get flatMap to work on the Actions-compatible ES version

**Pull request merged**